File: SUELDO-INTEGRADOR-ANGULAR/my-app/src/app/models/Producto.ts
import { Proveedor } from './Proveedor';

export interface ProductoForm {
  id?: number;
  sku: string;
  imagen: string;
  nombre: string;
  precio: number;
  descripcion: string;
  categoriaId: number;
  proveedorId: number;
}

export interface Producto {
  id: number;
  sku: string;
  imagen: string;
  nombre: string;
  precio: number;
  descripcion: string;
  categoria: Categoria;
  proveedor?: Proveedor;
}

export interface Categoria {
  id?: number;
  categoria: string;
  eliminado?: boolean;
}

File: SUELDO-INTEGRADOR-SPRING/integrador/src/main/java/com/asj/integrador/model/Producto.java
package com.asj.integrador.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "productos")
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String sku;
    private String imagen;
    private String nombre;
    private Double precio;
    @Column(length = 1024)
    private String descripcion;
    private boolean eliminado;
    @UpdateTimestamp
    @JsonFormat(pattern="dd/MM/yyyy HH:mm")
    private LocalDateTime fechaActualizacionRegistro;
    @CreationTimestamp
    @JsonFormat(pattern="dd/MM/yyyy HH:mm")
    private LocalDateTime fechaCreacionRegistro;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "proveedor_id", referencedColumnName = "id", nullable = false)
    private Proveedor proveedor;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id", referencedColumnName = "id", nullable = false)
    private Categoria categoria;
}
